datastore/postgres/migrations/matcher/14-v2-schema.sql

```diff
@@ -0,0 +1,585 @@
+-- vim: set foldmethod=marker :
+-- The formatting in this file is best-effort, all SQL formatters are garbage.
+-- It's dangerous to go alone! Take this.
+-- https://www.postgresql.org/docs/current/plpgsql.html
+CREATE SCHEMA matcher_v2;
+
+COMMENT ON SCHEMA matcher_v2 IS $$
+Matcher_v2 is a revised version of the schema for storing data to match against.
+
+Compared with the v1 structure this version normalizes more data, avoids some excessive write traffic pitfalls, and avoids client-side hashing schemes.
+$$;
+
+SET LOCAL search_path TO matcher_v2, "$user", public;
+
+-- {{{ Domain types
+CREATE TYPE Severity AS ENUM (
+	'Unknown',
+	'Negligible',
+	'Low',
+	'Medium',
+	'High',
+	'Critical'
+);
+
+COMMENT ON TYPE Severity IS 'Severity values are the defined Claircore "normalized" values.';
+-- {{{ VersionRange
+CREATE TYPE VersionRange AS RANGE (
+	SUBTYPE = TEXT[]
+	-- It would be great to be able to have a "canonical" function here that
+	-- handled digit-only strings transparently, but it's impossible:
+	-- 1. A shell type is a base type, so the user must be a superuser
+	-- 2. A PL cannot interact with shell types, so the function must be written in C.
+);
+
+COMMENT ON TYPE VersionRange IS 'VersionRange is a type for doing version comparisons in a standard way.';
+
+CREATE OR REPLACE FUNCTION version_check_array (vs TEXT[])
+	RETURNS BOOLEAN IMMUTABLE STRICT PARALLEL SAFE
+	LANGUAGE plpgsql
+	AS $$
+DECLARE
+	v TEXT;
+BEGIN
+	FOREACH v IN ARRAY vs LOOP
+		IF v ~ '\s' OR (v ~ '^\d+$' AND length(v) <> 10) THEN
+			RETURN FALSE;
+		END IF;
+	END LOOP;
+	RETURN TRUE;
+END;
+$$;
+
+COMMENT ON FUNCTION version_check_array IS 'This function reports whether a text array is well-formed to be used as a VersionRange';
+
+CREATE OR REPLACE FUNCTION version_check (mr matcher_v2.VersionMultiRange)
```

Comment:
Are we missing a `VersionMultiRange` definition here? Can't see to be able to run the migration without an error

Reply:
No, it's created automatically with a `RANGE` type: https://www.postgresql.org/docs/16/sql-createtype.html#SQL-CREATETYPE-RANGE